hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestDeleteWithSlowFollower.java

```diff
@@ -0,0 +1,274 @@
+    for (HddsDatanodeService dn : cluster.getHddsDatanodes()) {
+      if (ContainerTestHelper.isRatisFollower(dn, pipeline)) {
+        follower = dn;
+      } else if (ContainerTestHelper.isRatisLeader(dn, pipeline)) {
+        leader = dn;
+      }
+    }
+    Assert.assertNotNull(follower);
+    Assert.assertNotNull(leader);
+    //shutdown the slow follower
+    cluster.shutdownHddsDatanode(follower.getDatanodeDetails());
+    key.write(testData);
+    key.close();
+    XceiverClientSpi xceiverClient =
+        xceiverClientManager.acquireClient(pipeline);
+    ContainerProtos.ContainerCommandRequestProto.Builder request =
+        ContainerProtos.ContainerCommandRequestProto.newBuilder();
+    request.setDatanodeUuid(pipeline.getFirstNode().getUuidString());
+    request.setCmdType(ContainerProtos.Type.CloseContainer);
+    request.setContainerID(containerID);
+    request.setCloseContainer(
+        ContainerProtos.CloseContainerRequestProto.getDefaultInstance());
+    xceiverClient.sendCommand(request.build());
+    ContainerStateMachine stateMachine =
+        (ContainerStateMachine) ContainerTestHelper
+            .getStateMachine(leader, pipeline);
+    OmKeyArgs keyArgs = new OmKeyArgs.Builder().setVolumeName(volumeName).
+        setBucketName(bucketName).setType(HddsProtos.ReplicationType.RATIS)
+        .setFactor(HddsProtos.ReplicationFactor.THREE).setKeyName("ratis")
+        .build();
+    OmKeyInfo info = cluster.getOzoneManager().lookupKey(keyArgs);
+    BlockID blockID = info.getKeyLocationVersions().get(0).getLocationList().get(0)
+        .getBlockID();
+    OzoneContainer ozoneContainer;
+    final DatanodeStateMachine dnStateMachine =
+        leader.getDatanodeStateMachine();
+    ozoneContainer = dnStateMachine.getContainer();
+    KeyValueHandler keyValueHandler =
+        (KeyValueHandler) ozoneContainer.getDispatcher()
+            .getHandler(ContainerProtos.ContainerType.KeyValueContainer);
+    Container container =
+        ozoneContainer.getContainerSet().getContainer(blockID.getContainerID());
+    KeyValueContainerData containerData =
+        ((KeyValueContainerData) container.getContainerData());
+    long delTrxId = containerData.getDeleteTransactionId();
+    long numPendingDeletionBlocks = containerData.getNumPendingDeletionBlocks();
+    BlockData blockData =
+        keyValueHandler.getBlockManager().getBlock(container, blockID);
+    cluster.getOzoneManager().deleteKey(keyArgs);
+    GenericTestUtils.waitFor(() -> {
+      return
+          dnStateMachine.getCommandDispatcher().getDeleteBlocksCommandHandler()
+              .getInvocationCount() >= 1;
+    }, 500, 100000);
+    Assert.assertTrue(containerData.getDeleteTransactionId() > delTrxId);
+    Assert.assertTrue(
+        containerData.getNumPendingDeletionBlocks() > numPendingDeletionBlocks);
+    // make sure the chunk was never deleted on the leader even though
+    // deleteBlock handler is invoked
```

Comment:
I think we can do the above assertion check for all the nodes. The check can be moved into a function which can be invoked before and after restart restart of the follower.

Reply:
In the test one follower is shutdown before the container gets closed .The follower node which is already shutdown won't have the delete transaction log itself. Hence, once the dn starts up, blockDeleting service kicks off instantly and starts the first iteration of delete. Validating the deleteTransaction after restart may make the test flaky.